core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -597,9 +597,9 @@ public static String calculateX509Fingerprint(X509Certificate x509cert, String a
 			byte[] dataBytes = x509cert.getEncoded();
 			if (alg == null || alg.isEmpty() || alg.equals("SHA-1")|| alg.equals("sha1")) {
 				fingerprint = DigestUtils.sha1Hex(dataBytes);
-			} else if (alg.equals("SHA-256") || alg .equals("sha256")) {
```

Comment:
@pitbulk I have already tested in my local, that this does fix for my fingerprint validation check.